### PR TITLE
Clinical package generation

### PR DESCRIPTION
**Pull request by emilyrohrbough**
### Summary
Terra development has moved into two workspaces: terra-core and terra-clinical. Due to namespace formatting and package configurations, the generator needed updated to handle which repository the user was working in to generate appropriate package scaffolding. Issue #35 

### Updates
Upon running `yo terra-module`, the user is prompted with two messages:
1. `Terra repository:` which accepts `terra-core` or `terra-clinical`. It defaults to `terra-core` if no input is provided.
2. `Your module name:`, which is the desired name of the react component being created. Before the user would put `terra-detail-view`, now the user only needs to put `detail-view`. The first prompt handles the name space prefixes. 

@cerner/terra

### CODE REVIEW

**Comment by emilyrohrbough** [comment]
Issue #41 handled in this PR.

File: generators/app/templates/terra-package.json [hunk 1]
```diff
@@ -25,12 +25,15 @@
     "terra-toolkit": "^0.x"
   },
   "peerDependencies": {
-    "react": "^15.4.2",
-    "react-dom": "^15.4.2"
+    "react": "^15.5.4",
+    "react-dom": "^15.5.4",
+    "prop-types": "^15.5.8",
```
Comment (Matt-Butler) [inline generators/app/templates/terra-package.json]: I think we should move `prop-types` to a dependency.

https://github.com/cloudflare/cf-ui/blob/master/packages/cf-component-box/package.json#L17

Comment (Matt-Butler) [inline generators/app/templates/terra-package.json]: https://github.com/facebook/prop-types/issues/44

Comment (bjankord) [inline generators/app/templates/terra-package.json]: Agreed, lets make this update and then get this merged down.

File: generators/app/index.js [hunk 2]
```diff
@@ -58,9 +65,18 @@ module.exports = yeoman.Base.extend({
 
     var prompts = [{
       type: 'input',
-      name: 'projectName',
+      name: 'repository',
+      message: 'Terra repository:',
+      default: 'terra-core',
+      validate: function (input) {
+        var validRepos = ['terra-core', 'terra-clinical'];
```
Comment (ryanthemanuel) [inline generators/app/index.js]: Can we update the ReadMe for this generator to provide some guidance about what the user might expect when running it?

Comment (emilyrohrbough) [inline generators/app/index.js]: https://github.com/cerner/generator-terra-module/pull/40/commits/d0cf6adfd4c6639140fa74377e801b73576ae93c#diff-04c6e90faac2675aa89e2176d2eec7d8L20

File: generators/app/templates/terra-package.json (hunk 1 above)
Comment (emilyrohrbough) [inline generators/app/templates/terra-package.json]: [Changed Here](https://github.com/cerner/generator-terra-module/pull/40/commits/d0cf6adfd4c6639140fa74377e801b73576ae93c#diff-36443d7bec1e09c0ec1f9d7df8af48e1R36)

**Pull request merged**